common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/RestClientVrops.java

```diff
@@ -425,6 +459,61 @@ public PolicyDTO.Policy getDefaultPolicy() throws Exception {
 		return this.getAllPolicies().stream().filter(item -> item.getDefaultPolicy()).findFirst().orElse(null);
 	}
 
+	/**
+	 * Sets the priorities for a set of policies. Note that it applies to vROPs
+	 * 8.17.0 and later only. The order of the policy in the list define the policy
+	 * priorities. First policy in the list will be a policy with top priority.
+	 *
+	 * @param policies List of Policies.
+	 * @throws Exception exception if on or more policies cannot be found or
+	 *                   priority cannot be set.
+	 */
+	public void setPolicyPriorities(final List<String> policies) throws Exception {
+		// policy priority setting is available only since vROPs version 8.17.X
+		if (!this.isVersionAbove817()) {
+			return;
+		}
+		List<PolicyDTO.Policy> allPolicies = this.getAllPolicies();
+		List<String> policyIds = new ArrayList<String>();
+		List<String> missingPolicies = new ArrayList<String>();
+		policies.forEach(policyName -> {
+			PolicyDTO.Policy foundPolicy = allPolicies.stream().filter(item -> item.getName().equalsIgnoreCase(policyName)).findFirst().orElse(null);
+			if (foundPolicy == null) {
+				missingPolicies.add(policyName);
+			} else if (foundPolicy != null && foundPolicy.getDefaultPolicy()) {
+				// note that the default policy cannot be part of the policy ordering (according to the API documentation)
+				logger.warn("Skipping default policy '{}' from the policy priority order as it is not supported by vROPs", foundPolicy.getName());
+			} else {
+				policyIds.add(foundPolicy.getId());
+			}
+		});
+		if (!missingPolicies.isEmpty()) {
+			throw new RuntimeException(String.format("The policies '%s' cannot be found on the target system", this.concatenateList(missingPolicies, ", ")));
+		}
+
+		logger.info("Ordering policies by priority '{}'", this.concatenateList(policies, ", "));
+		Gson gson = new GsonBuilder().setLenient().create();
+		try {
+			UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromUri(new URI(getURIBuilder().setPath(POLICY_PRIORITY_PUBLIC_API).toString()));
+			URI restUri = uriBuilder.build().toUri();
+			HttpHeaders headers = new HttpHeaders();
+			headers.setContentType(MediaType.APPLICATION_JSON_UTF8);
+			Map<String, List<String>> parameters = new HashMap<String, List<String>>();
+			parameters.put("policyIds", policyIds);
+
+			ResponseEntity<String> response = restTemplate.exchange(restUri, HttpMethod.PUT, new HttpEntity<>(gson.toJson(parameters), headers), String.class);
+			if (response.getStatusCode() != HttpStatus.NO_CONTENT) {
+				throw new RuntimeException(response.getBody());
```

Comment:
Throwing an exception just to be caught and rethrown; better to declare the "response" variable before try, move the check & throw after the catch.